src/metabase/models/setting.clj

```diff
@@ -312,10 +317,18 @@
 ;;; |                                                defsetting macro                                                |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(def ^:private tag-for-type
+  {:string  String
+   :boolean Boolean
+   :integer Integer
+   :json    String
+   :double  Double})
+
 (defn metadata-for-setting-fn
   "Create metadata for the function automatically generated by `defsetting`."
   [{:keys [default description], setting-type :type, :as setting}]
   {:arglists '([] [new-value])
+   :tag (tag-for-type setting-type)
```

Comment:
Minor tweak to `defsetting` macro so a setting now has proper type metadata. e.g. 


```clojure
(defsetting my-cool-setting "Cool Setting")

(java.util.TimeZone/getTimeZone (metabase.driver/report-timezone))
```

would now work without reflection warnings.